ydb/services/metadata/manager/modification.h

```diff
@@ -152,15 +152,31 @@ template <class TObject>
 class TInsertObjectsActor: public TModifyObjectsActor<TObject> {
 private:
     using TBase = TModifyObjectsActor<TObject>;
+    bool ExistingOk = false;
 protected:
     virtual Ydb::Table::ExecuteDataQueryRequest BuildModifyQuery() const override {
         return TBase::Objects.BuildInsertQuery(TObject::GetBehaviour()->GetStorageTablePath());
     }
     virtual TString GetModifyType() const override {
         return "insert";
     }
+
+    void Handle(NRequest::TEvRequestFailed::TPtr& ev) override {
+        if (ev->Get()->GetStatus() == Ydb::StatusIds::PRECONDITION_FAILED && ExistingOk) {
+            NRequest::TDialogYQLRequest::TResponse resp;
+            this->Send(this->SelfId(), new NRequest::TEvRequestResult<NRequest::TDialogYQLRequest>(std::move(resp)));
+            this->Requests.clear(); // Remove history request
```

Comment:
Why do we remove all requests, if one fails?

Reply:
It is rather complicated code :) This second request is here: https://github.com/ydb-platform/ydb/blob/main/ydb/services/metadata/manager/modification.h#L93
It is history insertion and we don't need it in case we haven't inserted main thing.